infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftClientFactory.java

```diff
@@ -109,6 +113,15 @@ protected Config buildDefaultConfig(
     return theConfig;
   }
 
+  /**
+   * Builds the Openshift {@link Config} object based on a default {@link Config} object and an
```

Comment:
@davidfestal WDYT?

Reply:
> It doesn't use default config but rather provided. Javadocs and parameter name are misleading. Looks like the mistake was copied from the KubernetesClientFactory.

well the provided config __is a default one (based on the properties)__ that can be used as a basis to provided a new configuration customized for the workspaceId current userId). Afaict this comment was already there before this PR.

> So I would consider moving usage of the builder to the Kubernetes infra and client

I'd prefer do this in a next PR, because it also impacts Guice module configuration. And the involved design changes would require testing everything again.

> Please, add @Override annotation to this method

Sure